controller_manager/src/controller_manager.cpp

```diff
@@ -373,6 +373,14 @@ bool ControllerManager::shutdown_controllers()
 
 void ControllerManager::init_controller_manager()
 {
+  controller_manager_activity_publisher_ =
+    create_publisher<controller_manager_msgs::msg::ControllerManagerActivity>(
+      "~/activity", rclcpp::QoS(1).reliable().transient_local());
+  rt_controllers_wrapper_.set_on_switch_callback(
+    std::bind(&ControllerManager::publish_activity, this));
+  resource_manager_->set_on_component_state_switch_callback(
+    std::bind(&ControllerManager::publish_activity, this));
```

Comment:
I understand that this is easier to integrate now but shouldn't we simply make these part of the constructors?

Reply:
Sure, when we branch off, we will certainly do that :)